src/plugins/intel_cpu/src/emitters/plugin/x64/jit_bf16_emitters.hpp

```diff
@@ -33,56 +38,115 @@ class jit_uni_vcvtneps2bf16 : public jit_emitter {
         }
     }
 
+    template <typename Vmm>
+    void saturate_input(Vmm& truncated,
+                        const Vmm& in,
+                        const std::string& bf16_min_key,
+                        const std::string& bf16_max_key) const {
+        Vmm bf16_max = Vmm(aux_vec_idxs[0]);
+        Vmm bf16_min = Vmm(aux_vec_idxs[1]);
+        h->uni_vmovups(bf16_max, table_val(bf16_max_key));
+        h->uni_vmovups(bf16_min, table_val(bf16_min_key));
+        h->uni_vmaxps(truncated, in, bf16_min);
+        h->uni_vminps(truncated, truncated, bf16_max);
+    }
+
     template <dnnl::impl::cpu::x64::cpu_isa_t isa>
     void emit_isa(const std::vector<size_t>& in_vec_idxs, const std::vector<size_t>& out_vec_idxs) const {
         using namespace Xbyak;
-        using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41, Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
+        using Vmm = typename dnnl::impl::utils::
+            conditional3<isa == dnnl::impl::cpu::x64::sse41, Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
 
         Vmm in = Vmm(in_vec_idxs[0]);
 
-        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
-            Ymm out = Ymm(out_vec_idxs[0]);
-            h->vcvtneps2bf16(out, in);
-        } else if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx512_core) {
-            Zmm aux = Zmm(aux_vec_idxs[0]);
-            Zmm aux1 = Zmm(aux_vec_idxs[1]);
-            Ymm out = Ymm(out_vec_idxs[0]);
-
-            h->uni_vpsrld(aux, in, 16);
-            h->vpandd(aux, aux, table_val("one"));
-            h->uni_vmovups(aux1, table_val("even"));
-            h->uni_vpaddd(aux, aux1, aux);
-            h->uni_vpaddd(aux, in, aux);
-            h->vfixupimmps(aux, in, table_val("selector"), 0);
-            h->vpsrad(aux, aux, 16);
-            h->vpmovdw(out, aux);
-        } else if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::cpu_isa_t::avx2_vnni_2)) {
-            Xmm out = Xmm(out_vec_idxs[0]);
-            h->vcvtneps2bf16(out, in, PreferredEncoding::VexEncoding);
-        } else {  // round_to_nearest_even emulation
-            Vmm aux = Vmm(aux_vec_idxs[0]);
-            Xmm out = Xmm(out_vec_idxs[0]);
-
-            if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx2) {
-                h->uni_vandps(aux, in, table_val("rounding"));
+        if (mode_ == arithmetic_mode::saturation) {
+            Vmm truncated = Vmm(aux_vec_idxs[2]);
+            if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
+                Ymm out = Ymm(out_vec_idxs[0]);
+                saturate_input(truncated, in, "bf16_min", "bf16_max");
```

Comment:
It seems the same for all branches, move to outside?

Reply:
Done